src/Course/List.hs

```diff
@@ -200,26 +178,19 @@ flatten =
 -- prop> \x -> headOr x (flatMap id (y :. infinity :. Nil)) == headOr 0 y
 --
 -- prop> \x -> flatMap id (x :: List (List Int)) == flatten x
-flatMap ::
-  (a -> List b)
-  -> List a
-  -> List b
-flatMap =
-  error "todo: Course.List#flatMap"
+flatMap :: (a -> List b) -> List a -> List b
+flatMap f = flatten . P.fmap f
```

Comment:
Haha not supposed to use prelude. We already defined `map` above, right? you're allowed to use that! :)

`flatMap f = flatten . map f`

Reply:
yeah, I don't know why I did that hahah